src/Core/Data/SqlRedirectRepository.cs

```diff
@@ -0,0 +1,228 @@
+                    CreateStringParameter("newurl", entity.NewUrl),
+                    CreateIntParameter("state", entity.State),
+                    CreateBoolParameter("wildcardskipappend", entity.WildCardSkipAppend)),
+                    "An error occured while creating a redirect.");
+        }
+
+        private void Update(CustomRedirect entity)
+        {
+            var sqlCommand = $@"UPDATE {RedirectsTable}
+                                    SET OldUrl = @oldurl
+                                        ,NewUrl = @newurl
+                                        ,State = @state
+                                        ,WildCardSkipAppend = @wildcardskipappend
+                                    WHERE Id = @id";
+
+            ExecuteNonQuery(() =>
+                CreateCommand(
+                    sqlCommand,
+                    CreateGuidParameter("id", entity.Id.ExternalId),
+                    CreateStringParameter("oldurl", entity.OldUrl),
+                    CreateStringParameter("newurl", entity.NewUrl),
+                    CreateIntParameter("state", entity.State),
+                    CreateBoolParameter("wildcardskipappend", entity.WildCardSkipAppend)),
+                    "An error occured while updating a redirect.");
+        }
+
+        public void Delete(CustomRedirect entity)
+        {
+            var sqlCommand = $@"DELETE FROM {RedirectsTable}
+                                    WHERE Id = @id";
+
+            ExecuteNonQuery(() =>
+                CreateCommand(
+                    sqlCommand,
+                    CreateGuidParameter("id", entity.Id.ExternalId)),
+                    "An error occured while deleting a redirect.");
+        }
+
+        public CustomRedirect GetByOldUrl(string oldUrl)
+        {
+            var sqlCommand = $@"SELECT {AllFields} FROM {RedirectsTable}
+                                    WHERE OldUrl = @oldurl";
+
+            return ExecuteQuery(() =>
+                CreateCommand(
+                    sqlCommand,
+                    CreateStringParameter("oldurl", oldUrl)))
+                .FirstOrDefault();
+        }
+
+        public IEnumerable<CustomRedirect> GetAll()
+        {
+            var sqlCommand = $@"SELECT {AllFields} FROM {RedirectsTable}";
+
+            return ExecuteQuery(() => CreateCommand(sqlCommand));
+        }
+
+        public IEnumerable<CustomRedirect> GetByState(RedirectState state)
+        {
+            var sqlCommand = $@"SELECT {AllFields} FROM {RedirectsTable}
+                                    WHERE State = @state";
+
+            return ExecuteQuery(() =>
+                CreateCommand(
+                    sqlCommand,
+                    CreateIntParameter("state", (int)state)));
+        }
+
+        public IEnumerable<CustomRedirect> Find(string searchText)
+        {
+            var sqlCommand = $@"SELECT {AllFields} FROM {RedirectsTable}
+                                    WHERE OldUrl like '%' + @searchText + '%'
```

Comment:
Possible injection as not found url is controlled by end user.
Also sql repo should be protected against injection

Reply:
Nope, this is not an injection. @searchText is a SQL parameter, not a string.

Reply:
ok

Reply:
you ar right not a problem both searchText variables having same name confused me